doc/devbox_setup.md

```diff
@@ -0,0 +1,115 @@
+
+- Install Powershell https://github.com/PowerShell/PowerShell/blob/master/docs/installation/linux.md#ubuntu-1604
+- Install the latest .NET Core SDK CLI https://www.microsoft.com/net/learn/get-started/linux/ubuntu16-04
+
+<a name="advanced"/>
+
+## Optional Setup required to test Xamarin, Windows IoT and .NET Micro Framework
+
+### Installing Visual Studio 2017 for Xamarin applications
+
+- Install [Visual Studio 2017][visual-studio]. You can use the **Visual Studio Community** Free download if you meet the licensing requirements.
+- During the installation of Visual Studio 2017, we found that selecting the following workloads and components helps in running and debugging the SDK.
+
+Workloads:
+![](./workloads.png)
+
+Components:
+![](./components.png)
+
+### Installing Windows IoT Core SDK
+
+Install the Microsoft IoT Windows Core Project Templates for Visual Studio 2017 from the Extension Marketplace: 
+	https://marketplace.visualstudio.com/items?itemName=MicrosoftIoT.WindowsIoTCoreProjectTemplatesforVS15
+
+### Installing .NET Micro Framework
+
+#### 1. Download and Install Visual Studio Community 2015 Update 3 (approx. size: 9GB)
+
+NetMF is not supported on VS2017. We recommend a separate virtual machine with VS2015.
+VS2015 Community is sufficient. Download from: https://www.visualstudio.com/vs/older-downloads/
+
+Under Programming Languages, Visual C++ make sure that `Common Tools for Visual C++ 2015` is selected.
+
+#### 2. Download and install .NET MicroFramework from https://netmf.github.io/
+
+1. SDK v 4.4: 
+  * MicroFrameworkSDK.msi  (SDK v4.4) - Complete Installation
+  * NetMFVS14.vsx
+		
+2. SDK 4.3 - The official site has been archived. Either:
+  * Download the Codeplex archive (3.4GB) then unpack to find the MSI:
+    * Under releases\13 unpack c4ad01a9-68f6-48cd-82e5-e6d154b0eb2f
+    * Rename c4ad01a9-68f6-48cd-82e5-e6d154b0eb2f to netmf-v4.3.2-SDK-QFE2-RTM.zip and unzip
+    * Install MicroFrameworkSDK.msi - Complete Installation
+  * Download from http://developer.wildernesslabs.co/Netduino/About/Downloads/
+  
+3. Install .NET Framework 3.5
+
+<a name="testprereq"/>
+
+## Test prerequisites
+
+Each test project that requires additional changes to your machine contains a `prerequisites` folder. Please follow the instructions within that folder on how to set-up your machine in order to successfully run our tests.
+
+#### iothub/service/prerequisites one-time setup (Windows Only)
+Run the following as admin: `iothub\service\tests\prerequisites\windows_install.cmd`
+
+#### e2e/test/prerequisites
+
+1. Create copies of the prerequisite scripts in a separate folder (optionally include this folder in your PATH environment variable for easy access. 
+
+    * Windows: copy `e2e/test/prerequisites/iot_config.cmd_template` to a folder in your PATH `iot_config.cmd`
+    * Linux/OSX: copy `e2e/test/prerequisites/iot_config.sh_template` to `iot_config.sh`
+
+**Do not** check-in these files after you have edited them as they contain
```

Comment:
We need to mention that they contain user specific values

Reply:
Good catch! Thanks... looks like I got interrupted in the middle of this sentence.